From 8642fd5c077ff27eeb0fae4b7ffcfb929e9e511e Mon Sep 17 00:00:00 2001
From: Klaus Zimmermann <klaus.zimmermann@smhi.se>
Date: Fri, 3 Feb 2023 13:22:24 +0100
Subject: [PATCH 6/7] Make pathtype recognize Windows style absolute paths

---
 scripts/pathtype | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/scripts/pathtype b/scripts/pathtype
index 588b008ca1..c297811699 100755
--- a/scripts/pathtype
+++ b/scripts/pathtype
@@ -1,8 +1,9 @@
 #!/bin/sh
 # return "abs" if input path absolute and "rel" if input path relative
 firstchar=`echo $1 | awk '{ print substr($0,1,1) }'`
-if ([ $firstchar = "/" ]); then
-echo "abs"
+secondchar=`echo $1 | awk '{ print substr($0,2,1) }'`
+if [[ $firstchar = "/" || $secondchar = ":" ]]; then
+    echo "abs"
 else
-echo "rel"
+    echo "rel"
 fi
-- 
2.39.1

