diff -ruN --no-dereference esmf-8.7.0/scripts/pathtype esmf-8.7.0-patch/scripts/pathtype
--- esmf-8.7.0/scripts/pathtype	2024-10-03 21:21:21.000000000 -0600
+++ esmf-8.7.0-patch/scripts/pathtype	2024-10-26 11:50:13.260479971 -0600
@@ -1,8 +1,10 @@
-#!/bin/sh
+#!/bin/bash
 # return "abs" if input path absolute and "rel" if input path relative
 firstchar=`echo $1 | awk '{ print substr($0,1,1) }'`
-if ([ $firstchar = "/" ]); then
-echo "abs"
+secondchar=`echo $1 | awk '{ print substr($0,2,1) }'`
+if [[ $firstchar = "/" || $secondchar = ":" ]]; then
+    echo "abs"
 else
-echo "rel"
+    echo "rel"
 fi
+
