diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS esmf-8.4.0.deactivate-system-tests/build/common.mk esmf-8.4.0/build/common.mk
--- esmf-8.4.0.deactivate-system-tests/build/common.mk	2022-12-02 09:14:43.751840679 +0100
+++ esmf-8.4.0/build/common.mk	2022-12-02 09:18:36.110095673 +0100
@@ -2449,7 +2449,7 @@
 	@if [ $(ESMF_COMM) = "mpiuni" ] ; then \
 	  $(MAKE) clean_check $(TEST_TARGETS_UNI) results_ck_summary ;\
 	else \
-	  $(MAKE) clean_check $(TEST_TARGETS) results_ck_summary ;\
+	  @echo "Skipping all tests" ;\
         fi
 
 
