diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS esmf-8.4.0.long-running-tests-removed/build/common.mk esmf-8.4.0/build/common.mk
--- esmf-8.4.0.long-running-tests-removed/build/common.mk	2022-12-02 09:12:48.858225390 +0100
+++ esmf-8.4.0/build/common.mk	2022-12-02 09:13:58.363585530 +0100
@@ -2415,8 +2415,7 @@
 # vars used below in the all_tests target, because these are in the pattern
 # (build, run), (build, run), ... not (build, build, ...) then (run, run, ...)
 
-TEST_TARGETS = build_unit_tests run_unit_tests \
-               build_system_tests run_system_tests
+TEST_TARGETS = build_unit_tests run_unit_tests
 
 ALLTEST_TARGETS = $(TEST_TARGETS) \
                   build_examples run_examples
